includes/DomainCheckJob.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Miraheze\RequestSSL;
+
+use GenericParameterJob;
+use Job;
+use MediaWiki\MediaWikiServices;
+use User;
+
+class DomainCheckJob extends Job implements GenericParameterJob {
+	public function __construct( array $params ) {
+		parent::__construct( 'DomainCheckJob', $params );
+	}
+	public function run() {
+		$mwServices = MediaWikiServices::getInstance();
+		$requestSslManager = $mwServices->get( 'RequestSSLManager' );
+		$requestSslManager->fromID( $this->params['requestID'] );
+		$customDomain = parse_url( $requestSslManager->getCustomDomain(), PHP_URL_HOST );
+		if ( !$customDomain ) {
+			// Custom domain does not have a hostname, bail out.
+			// TODO: Log an exception.
```

Comment:
```suggestion
			$this->setLastError( 'Custom domain does not have a hostname.' );
```

Exceptions should never be directly logged in a job or the job won't fail it'll just break it and cause issues with other jobs also, causing it to infinitely loop due to retries, unless you use this, then it would just be abandoned, but still setLastError is better:

```php
/**
 * @return bool
 */
public function allowRetries(): bool {
	return false;
}
```

Reply:
Done in [776dd4f](https://github.com/miraheze/RequestSSL/pull/40/commits/776dd4f365877070992f63a8b3339e58cb3302bc)